File: Contests/21/6.py
import itertools
from collections import Counter

hat = [0] * n
tshirt = [0] * n
pants = [0] * n

n = int(input())

for _ in range(n):
    ti, ci = map(int, input().split())
    
    if ti == 1:
        hat[ci] += 1
    elif ti == 2:
        tshirt[ci] += 1
    elif ti == 3:
        pants[ci] += 1

result = 0

for (ch, counth), (ct, countt), (cp, countp) in itertools.product(hat.items(), tshirt.items(), pants.items()):
    if ct != ch and cp != ct and cp != ch:
        result += counth * countp * countt

print(result)

File: Contests/21/6.cpp
#include <iostream>
#include <map>
#include <vector>

using namespace std;

int main() {
    map<int, int> hat;
    map<int, int> tshirt;
    map<int, int> pants;

    int n;
    cin >> n;

    for (int i = 0; i < n; i++) {
        int ti, ci;
        cin >> ti >> ci;

        if (ti == 1) {
            hat[ci]++;
        } else if (ti == 2) {
            tshirt[ci]++;
        } else if (ti == 3) {
            pants[ci]++;
        }
    }

    int result = 0;

    for (auto& h : hat) {
        for (auto& t : tshirt) {
            for (auto& p : pants) {
                if (t.first != h.first && p.first != t.first && p.first != h.first) {
                    result += h.second * t.second * p.second;
                }
            }
        }
    }

    cout << result << endl;

    return 0;
}
